pulsar-functions/proto/src/main/proto/Function.proto

```diff
@@ -141,6 +141,11 @@ message FunctionMetaData {
     uint64 version = 3;
     uint64 createTime = 4;
     map<int32, FunctionState> instanceStates = 5;
+    FunctionAuthenticationSpec functionAuthSpec = 6;
+}
+
+message FunctionAuthenticationSpec {
+    bytes data = 1;
```

Comment:
Does `data` includes the credential? Or a pointer to the credentials? Can you add comment to clarify the field meaning?
If it includes credentials, should this be included in the function metadata?

Reply:
So it can include the actual credentials (not secure) or a pointer to the credentials.  This all depends on the underlying implementation of the interface.  I left this field open ended so that there can be flexibility of what can be stored in here

Reply:
I think it should always include the credentials. If you want to make it secure, make it secure using the caller of cacheAuthData, not within cacheAuthData

Reply:
@ivankelly I don't quite follow what you are proposing.  This protobuf message will be stored in the function metadata topic.  This message was designed with a generic purpose to store some data for the function authentication provider so that it can distribute it to all the workers.  What that data is is up to the function authentication provider implementation.  This gives us flexibility in the types of authentication we can support in the future.